### PR TITLE
Add responsive mobile navigation bar

### PR DESCRIPTION
**Pull request by chownces**
### Description

- Added responsive mobile navigation bar
- Updated Playground Introduction tab icon from 'COMPASS' to 'HOME' as 'COMPASS' will be used for Navigation Bar UI
- Added new Jest Enzyme tests for the new NavigationBarMobileSideMenu component
- Updated Jest snapshots for NavigationBar and AcademyNavigationBar
- Added dependency 'react-responsive' and dev dependency '@types/react-responsive'

###### Playground Only (Mobile Breakpoint)
Previously:
image
Now:
image
<br>
###### Academy while logged out (Mobile Breakpoint)
Previously:
image

Now:
(side menu closed)
image
(side menu opened)
image
<br>
###### Academy while logged in (Mobile Breakpoint)
Previously:
image

Now:
(side menu closed)
image
(side menu opened)
image
<br>
#### Changes to UI (Playground Introduction Tab)
Previously:
image

Now:
image

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Code quality improvements

### How to test

- Tested with browser resizing on MacOS Big Sur with latest versions of Chrome, Firefox and Safari
- Tested mobile navigation bar on iOS14 with latest versions of Chrome and Safari

### Checklist

- [x] I have tested this code
- [ ] I have updated the documentation

### CODE REVIEW

**Comment by chownces** [comment]
1. I have excluded links to staff routes (Dashboard, Ground Control, Sourcereel, Grading, Story Simulator) in the mobile navigation bar side menu for now. Should they be included?

2. Should '/pages/academy/subcomponents/AcademyNavigationBar.tsx' be consolidated under '/commons/NavigationBar'? I have updated the Academy Navigation Bar such that it can now be accessible from anywhere in the application when logged in (it was previously only accessible in the '/academy/*' url).

**Review by caipng — COMMENTED** [review]
looks good to me

File: src/commons/navigationBar/NavigationBar.tsx [hunk 1]
```diff
@@ -92,7 +156,23 @@ const NavigationBar: React.SFC<NavigationBarProps> = props => (
...
+  );
+
+  return (
+    <>
+      <Navbar className={classNames('NavigationBar', 'primary-navbar', Classes.DARK)}>
+        {Constants.playgroundOnly
+          ? playgroundOnlyNavbarLeft
+          : isMobile
+          ? mobileNavbarLeft
+          : desktopNavbarLeft}
```
[the remark this answers is not in the document]
Comment (martin-henz) [reply src/commons/navigationBar/NavigationBar.tsx:168]: For what it's worth, this is the indentation of conditional expressions we have adopted in SICP JS, for example see: https://source-academy.github.io/sicp/chapters/4.1.1.html#p4
(not saying that this is standard JS indentation, just that we found this suitable for SICP JS)
(current coding style for cadet-frontend doesn't say anything about conditional expressions: https://source-academy.github.io/sicp/chapters/4.1.1.html#p4)

**Comment by martin-henz** [comment]
> 1. I have excluded links to staff routes (Dashboard, Ground Control, Sourcereel, Grading, Story Simulator) in the mobile navigation bar side menu for now. Should they be included?
> 2. Should '/pages/academy/subcomponents/AcademyNavigationBar.tsx' be consolidated under '/commons/NavigationBar'? I have updated the Academy Navigation Bar such that it can now be accessible from anywhere in the application when logged in (it was previously only accessible in the '/academy/*' url).

Regarding (1): yes, let's exclude the staff interface from the scope of the mobile version, indefinitely.

(Regarding (2): don't understand. will discuss off-line)

File: src/commons/navigationBar/NavigationBar.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (caipng) [reply src/commons/navigationBar/NavigationBar.tsx:168]: ah I see, maybe it was just me who got a bit confused. the [SICP style guide](https://source-academy.github.io/source/source_styleguide.pdf) do seem to encourage indentation:

but nevertheless its a nonissue

[the remark this answers is not in the document]
Comment (martin-henz) [reply src/commons/navigationBar/NavigationBar.tsx:168]: Thanks for bringing it up. I'll add the SICP JS style to our Source style guide.

[the remark this answers is not in the document]
Comment (chownces) [reply src/commons/navigationBar/NavigationBar.tsx:168]: `yarn format` will format it as per the commit and remove this additional indentation

[the remark this answers is not in the document]
Comment (caipng) [reply src/commons/navigationBar/NavigationBar.tsx:168]: seems that this is quite a [hotly debated issue regarding prettier](https://github.com/prettier/prettier/issues/5814) but seems that for now there we are stuck with 'flat' tenaries and no options to change it

[the remark this answers is not in the document]
Comment (caipng) [reply src/commons/navigationBar/NavigationBar.tsx:168]: seems like this is an intentional (unconfigurable) feature of prettier and is a hotly debated issue (see prettier repo issue 5814). looks like we will be stuck with these 'flat' tenaries for now

File: src/commons/navigationBar/NavigationBarMobileSideMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+    <NavLink
+      activeClassName={Classes.ACTIVE}
+      className={classNames(
+        'NavigationBar__link__mobile',
+        Classes.BUTTON,
+        Classes.MINIMAL,
+        Classes.LARGE
+      )}
+      to="/sourcecast/:sourcecastId?"
```
[the remark this answers is not in the document]
Comment (chownces) [reply src/commons/navigationBar/NavigationBarMobileSideMenu.tsx]: thanks for spotting this! updated

**Review by caipng — COMMENTED** [review]
looks good!

**Review by martin-henz — APPROVED** [review]
as reviewed by caipng

**Pull request merged**